utils/mgmt_system.py

```diff
@@ -71,6 +88,19 @@ def get_search_filter_spec(self, *args, **kwargs):
 
         return pfs
 
+    def __getattribute__(self, attr):
+        # fetch the attribute using parent class to avoid recursion
+        res = super(_PsphereClient, self).__getattribute__(attr)
+        # any callable (except 'login') is protected against unexpected logout
+        if callable(res) and attr != 'login':
+            try:
+                return self._cached_retry[attr]
```

Comment:
The recursion must be caused by something else because `_cached_retry` is a non-callable (dict)... looking into it.